dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/advanced/DateRangeValidator.java

```diff
@@ -70,14 +69,20 @@ private boolean isValidDateRange(LocalDate dateFrom, LocalDate dateTo, String da
         return isValid;
     }
 
-    private void isDateValueCorrect(Object value) throws ParseException {
-        DateUtils.parseDateStrictly(value.toString(), DATE_FORMATS);
+    private Either<String, Boolean> isDateValueCorrect(Object value) {
+        try {
+            DateUtils.parseDateStrictly(value.toString(), DATE_FORMATS);
+        } catch(ParseException  pe) {
+            return Either.left("Invalid date inserted. Given value: " + value.toString() + ". Acceptable formats: " + DATE_FORMATS);
+        }
+        return Either.right(true);
     }
 
-    private void isDatePatternCorrect(Object value) throws ParseException {
+    private Either<String, Boolean> isDatePatternCorrect(Object value) {
         if(!value.toString().matches(DATE_PATTERN)) {
```

Comment:
Suggestion: in that kind of methods with simple if-else one could use `? : ` operator instead.